tests/johnsnowlabs/test_jsl_api.py

```diff
@@ -0,0 +1,782 @@
+import json
+import os
+from pathlib import Path
+from unittest import mock
+
+import numpy as np
+import pandas as pd
+import pyspark
+import pytest
+import yaml
+from johnsnowlabs import nlp
+from packaging.version import Version
+
+import mlflow
+import mlflow.johnsnowlabs
+import mlflow.tracking
+import mlflow.utils.file_utils
+from mlflow import pyfunc
+from mlflow.environment_variables import MLFLOW_DFS_TMP
+from mlflow.johnsnowlabs import _add_code_from_conf_to_system_path
+from mlflow.models import Model, infer_signature, build_docker
+from mlflow.models.utils import _read_example
+from mlflow.pyfunc import spark_udf
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils.model_utils import _get_flavor_configuration
+from tests.helper_functions import (
+    _compare_conda_env_requirements,
+    _assert_pip_requirements,
+    _compare_logged_code_paths,
+    _mlflow_major_version_string, )
+from tests.helper_functions import score_model_in_sagemaker_docker_container
+
+# TODO you must update this
+license_keys = {
+    "AWS_ACCESS_KEY_ID": None,
+    "AWS_SECRET_ACCESS_KEY": None,
+    "SPARK_NLP_LICENSE": None,
+    "SECRET": None}
+
+os.environ.update(license_keys)
+
+# Before running any tests make sure the following environment variables are set:
+
+if not license_keys['SECRET']:
```

Comment:
Why is this here? Either we're setting the key as part of CI config from a secure key vault location to actually use the service, or we're mocking this. Is this ever going to be reachable?

Reply:
@BenWilson2  This is meant for developers/CI to make sure secrets are defined and show whats wrong. 
I will use validate_env_vars() from `mlflow.johnsnowlabs` instead here